mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -2122,37 +2224,31 @@ private void updateAndSaveJobCluster(IJobClusterMetadata jobCluster) throws Exce
      */
     /*package protected*/
     private Optional<JobDefinition> getLastSubmittedJobDefinition(final List<JobInfo> existingJobsList,
-                                                                  final List<CompletedJob> completedJobs, Optional<JobDefinition> jobDefinitionOp,
+                                                                  final List<CompletedJob> completedJobs,
+                                                                  Optional<JobDefinition> jobDefinitionOp,
                                                                   MantisJobStore store) {
         if(logger.isTraceEnabled()) { logger.trace("Entering getLastSubmittedJobDefinition"); }
         if(jobDefinitionOp.isPresent()) {
             return jobDefinitionOp;
         }
 
-        Optional<JobId> lastJobId = JobListHelper.getLastSubmittedJobId(existingJobsList,completedJobs);
+        Optional<JobId> lastJobId = JobListHelper.getLastSubmittedJobId(existingJobsList, completedJobs);
         if(lastJobId.isPresent()) {
-            Optional<JobInfo> jobInfoForNonTerminalJob = jobManager.getJobInfoForNonTerminalJob(lastJobId.get());
-            if(jobInfoForNonTerminalJob.isPresent()) {
-                if(logger.isTraceEnabled()) { logger.trace("Exit getLastSubmittedJobDefinition {}", jobInfoForNonTerminalJob.get().jobDefinition); }
-                return of(jobInfoForNonTerminalJob.get().jobDefinition);
-            } else {
-                Optional<CompletedJob> completedJob = jobManager.getCompletedJob(lastJobId.get());
-                if(completedJob.isPresent()) {
-                    try {
-                        Optional<IMantisJobMetadata> archivedJob = store.getArchivedJob(completedJob.get().getJobId());
-                        if(archivedJob.isPresent()) {
-                            if(logger.isTraceEnabled()) { logger.trace("Exit getLastSubmittedJobDefinition returning job {} with defn {}", archivedJob.get().getJobId(), archivedJob.get().getJobDefinition()); }
-                            return of(archivedJob.get().getJobDefinition());
-                        } else {
-                            logger.warn("Could not find load archived Job {} for cluster {}", completedJob.get().getJobId(), name);
-                        }
-                    } catch (Exception e) {
-                        logger.warn("Archived Job {} could not be loaded from the store due to {} ", completedJob.get().getJobId(), e.getMessage());
+            Optional<CompletedJob> completedJob = jobManager.getCompletedJob(lastJobId.get());
```

Comment:
why is this changed to completedJob?

Reply:
This is the quick submit path and for jobs with active job actors, it's passed to job actors instead of reaching here.
I will update the java doc on this method.